### PR TITLE
#6158 - feat: making sure created and updated date appears on certain…

### PR DESCRIPTION
**Pull request by DipakShrestha18**
… content types.

### CODE REVIEW

File: src/components/ResultsItem/ResultsItem.tsx [hunk 1]
```diff
@@ -19,6 +19,7 @@ export type ResultItemProps = {
   meta?: ResultItemMetaProps;
   text?: string;
   title: string;
+  nodeType?: string;
   type?: 'content' | 'file' | 'taxonomy_term';
```
Comment (chriddell) [inline src/components/ResultsItem/ResultsItem.tsx:23]: wondering if the naming of these two parameters is confusing now... what would `type` (content/file/taxonomy_term) be called in Drupal? File type? Content type?

File: src/components/ResultsItem/ResultsItem.tsx [hunk 2]
```diff
@@ -37,15 +39,18 @@ export const ResultsItem = ({
...
           &nbsp;
         </span>
-        <span className="result-item__meta-date">
-          <FormattedDate dateString={meta.date} />
-          &nbsp;
-        </span>
-        {meta.lastUpdated && (
-          <span className="result-item__meta-last-updated">
-            Updated <FormattedDate dateString={meta.lastUpdated} />
+        {meta.date && (nodeType === 'article' || nodeType === 'press_release') && (
```
Comment (chriddell) [inline src/components/ResultsItem/ResultsItem.tsx]: How about:

```js
meta.data && ['article', 'press_release'].includes(nodeType)
```

instead? (more succinct, also more extensible)

File: src/components/ResultsItem/ResultsItem.tsx [hunk 3]
```diff
@@ -37,15 +39,18 @@ export const ResultsItem = ({
...
+          <span className="result-item__meta-date">
+            <FormattedDate dateString={meta.date} />
+            &nbsp;
           </span>
         )}
+        {meta.lastUpdated &&
+          (nodeType === 'article' || nodeType === 'press_release') && (
```
Comment (chriddell) [inline src/components/ResultsItem/ResultsItem.tsx]: Same as above, how about:

```js
['article', 'press_release'].includes(nodeType)
```

File: src/components/ResultsItem/ResultsItem.tsx (hunk 1 above)
Comment (DipakShrestha18) [reply src/components/ResultsItem/ResultsItem.tsx:23]: I'm trying to be really specific and trying to align with what you get from solr as well, that's why I called it nodeType.

Comment (chriddell) [reply src/components/ResultsItem/ResultsItem.tsx:23]: yeah with you there, wondering more about the name of the `type` parameter now that we have to "type" params...

Comment (chriddell) [reply src/components/ResultsItem/ResultsItem.tsx:23]: would it be better as `contentType`? or `fileType`?

Comment (chriddell) [reply src/components/ResultsItem/ResultsItem.tsx:23]: actually let's park this and address is another PR (if needs be)

**Review by chriddell — CHANGES_REQUESTED** [review]
ignore type naming thing - see array.includes method only

File: src/components/ResultsItem/ResultsItem.tsx (hunk 2 above)
Comment (DipakShrestha18) [reply src/components/ResultsItem/ResultsItem.tsx]: neat!! done.

**Review by chriddell — APPROVED** [review]

**Pull request merged**